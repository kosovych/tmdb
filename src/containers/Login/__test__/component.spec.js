import { shallow } from 'enzyme'
import React from 'react'
import LoginComponent from '../component'

describe('Login Component', () => {
  const requiredProps = {
    loading: false,
    handleSubmit: jest.fn()
  }

  describe('with error', () => {
    it('should matches snapshot', () => {
      const container = shallow(<LoginComponent {... requiredProps} />)
      expect(container).toMatchSnapshot()
    })
  })

  describe('without error', () => {
    it('should matches snapshot', () => {
      const defaultProps = {
        errorMessage: 'Error'
      }
      const container = shallow(<LoginComponent
        {... requiredProps}
        {...defaultProps}
      />)
      expect(container).toMatchSnapshot()
    })
  })
})
