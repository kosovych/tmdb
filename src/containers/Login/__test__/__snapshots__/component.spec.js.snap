// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component with error should match snapshot 1`] = `
<div
  className="center"
>
  <Layout>
    <Content>
      <Row
        justify="center"
        type="flex"
      >
        <Col
          lg={
            Object {
              "offset": 2,
              "span": 8,
            }
          }
          sm={
            Object {
              "offset": 1,
              "span": 10,
            }
          }
        >
          <Form
            onSubmit={[MockFunction]}
          >
            <Title>
              The Movie DB
            </Title>
            <Field
              component={[Function]}
              name="username"
              placeholder="Username"
              prefix={
                <ForwardRef(UserOutlined)
                  style={
                    Object {
                      "color": "rgba(0,0,0,.25)",
                    }
                  }
                />
              }
            />
            <Field
              component={[Function]}
              name="password"
              placeholder="Password"
              prefix={
                <ForwardRef(LockOutlined)
                  style={
                    Object {
                      "color": "rgba(0,0,0,.25)",
                    }
                  }
                />
              }
              type="password"
            />
            <Button
              htmlType="submit"
              loading={false}
              type="primary"
            >
              Log in
            </Button>
          </Form>
        </Col>
      </Row>
    </Content>
  </Layout>
</div>
`;

exports[`Login Component without error should match snapshot 1`] = `
<div
  className="center"
>
  <Layout>
    <Content>
      <Row
        justify="center"
        type="flex"
      >
        <Col
          lg={
            Object {
              "offset": 2,
              "span": 8,
            }
          }
          sm={
            Object {
              "offset": 1,
              "span": 10,
            }
          }
        >
          <Form
            onSubmit={[MockFunction]}
          >
            <Title>
              The Movie DB
            </Title>
            <Alert
              message="Error"
              type="error"
            />
            <Field
              component={[Function]}
              name="username"
              placeholder="Username"
              prefix={
                <ForwardRef(UserOutlined)
                  style={
                    Object {
                      "color": "rgba(0,0,0,.25)",
                    }
                  }
                />
              }
            />
            <Field
              component={[Function]}
              name="password"
              placeholder="Password"
              prefix={
                <ForwardRef(LockOutlined)
                  style={
                    Object {
                      "color": "rgba(0,0,0,.25)",
                    }
                  }
                />
              }
              type="password"
            />
            <Button
              htmlType="submit"
              loading={false}
              type="primary"
            >
              Log in
            </Button>
          </Form>
        </Col>
      </Row>
    </Content>
  </Layout>
</div>
`;
